app/services/expired_registration_cleanup_service.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+class ExpiredRegistrationCleanupService < ::WasteExemptionsEngine::BaseService
+  def run
+    return unless registration_ids.any?
+
+    ActiveRecord::Base.transaction do
+      # paper_trail will try creating a record for the deletion
+      # unless we execute within this block
+      PaperTrail.request(enabled: false) do
+        clear_registrations!
+      end
+      clear_versions!
+      clear_versions_archive!
+    end
+  end
+
+  private
+
+  # registration exemptions that were deregistered (ceased or revoked)
+  # or expired over 7 years ago
+  def registration_ids
+    @registration_ids ||=
+      WasteExemptionsEngine::RegistrationExemption
+      .where("deregistered_at <= ?", date)
```

Comment:
`deregistered_at` is of type date, `updated_at` is of type datetime and `date` is of type string. Is this query robust?